### PR TITLE
feat: added emmetio implicit-tag definition

### PR DESCRIPTION
**Pull request by rfverbruggen**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## emmetio__implicit-tag ([<kbd>unpkg</kbd>](https://unpkg.com/browse/emmetio__implicit-tag@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'emmetio__implicit-tag'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7bf15e456081f431974eeb0bcfe4b00e91e29f18
</p>

File: types/emmetio__implicit-tag/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Type definitions for @emmetio/implicit-tag 1.0
+// Project: https://github.com/emmetio/implicit-tag#readme
+// Definitions by: Robbert Verbruggen <https://github.com/rfverbruggen>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare module '@emmetio/implicit-tag' {
+    export default function resolveImplicitName(parentName: string): string;
+}
```
Comment (rbuckton) [inline types/emmetio__implicit-tag/index.d.ts:8]: You shouldn't need to use `declare module` here.

```suggestion
export default declare function resolveImplicitName(parentName: string): string;
```

Comment (rbuckton) [reply types/emmetio__implicit-tag/index.d.ts:8]: Though it looks like this should probably be the following intead:

```suggestion
declare function resolveImplicitName(parentName: string): string;
export = resolveImplicitName;
```

According to https://unpkg.com/browse/@emmetio/implicit-tag@1.0.0/dist/implicit-tag.cjs.js#L33, the export uses `module.exports = ...` which translates to `export = ...` in TypeScript.

The package has an ES Module output as well, but it's only referenced using the [`"module"` property in package.json](https://unpkg.com/browse/@emmetio/implicit-tag@1.0.0/package.json#L6), which TypeScript doesn't recognize.

File: types/emmetio__implicit-tag/tslint.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+    "extends": "@definitelytyped/dtslint/dt.json",
+    "rules": {
+        "no-single-declare-module": false,
```
Comment (rbuckton) [inline types/emmetio__implicit-tag/tslint.json:4]: This shouldn't be necessary if you apply the other suggested changes:
```suggestion
```

File: types/emmetio__implicit-tag/tslint.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+                        false
+                    ],
+                    [
+                        "NeedsExportEquals",
+                        false
+                    ]
+                ]
+            }
+        ]
+    }
```
Comment (rbuckton) [inline types/emmetio__implicit-tag/tslint.json:21]: It looks like the lint errors are correct. The file that the `"main"` property in package.json points to is a CommonJS file that uses `module.exports = ...`.

```suggestion
```

**Review by rbuckton — CHANGES_REQUESTED** [review]

File: types/emmetio__implicit-tag/tslint.json (hunk 3 above)
Comment (rfverbruggen) [reply types/emmetio__implicit-tag/tslint.json]: Implemented your suggestions

File: types/emmetio__implicit-tag/tslint.json (hunk 2 above)
Comment (rfverbruggen) [reply types/emmetio__implicit-tag/tslint.json]: Implemented your suggestions

File: types/emmetio__implicit-tag/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictFunctionTypes": true,
+        "strictNullChecks": true,
+        "baseUrl": "../",
+        "typeRoots": [
+            "../"
+        ],
+        "types": [],
```
[the remark this answers is not in the document]
Comment (rfverbruggen) [reply types/emmetio__implicit-tag/tsconfig.json:15]: Implemented your suggestions

File: types/emmetio__implicit-tag/index.d.ts (hunk 1 above)
Comment (rfverbruggen) [reply types/emmetio__implicit-tag/index.d.ts]: Implemented your suggestions

**Review by rbuckton — APPROVED** [review]

**Comment by rfverbruggen** [comment]
Ready to merge

**Pull request merged**